packages/core/src/errors.js

```diff
@@ -3,6 +3,38 @@
 const util = require('util');
 const _ = require('lodash');
 
+class AppError extends Error {
+  constructor(message, code, status) {
+    super(
+      JSON.stringify({
+        message,
+        code,
+        status
+      })
+    );
+    this.name = 'AppError';
+  }
+}
+
+class ResponseError extends Error {
+  constructor(response) {
+    super(
+      JSON.stringify({
+        status: response.status,
+        headers: {
+          'content-type': response.headers.get('content-type')
+        },
+        content: response.content,
+        request: {
+          url: response.request.url
+        }
+      })
+    );
+    this.name = 'ResponseError';
+    this.contextify = false;
+  }
+}
+
 // Make some of the errors we'll use!
 const createError = name => {
   const NewError = function(message = '') {
```

Comment:
I'm not sure why we use this pattern, but it makes it impossible to do:

```
const error = new z.errors.HaltedError();
assert error instanceof z.errors.HaltedError
```

as `error` will be an instance of the `NewError` function here.

Reply:
Used mostly so that we didn't need to write `class X extends Error {` over and over. But we can probably clean this up with newer JS features.

Reply:
A good one for a separate PR - would be nice to then be able to use `instanceof` on all of them.